### ISSUE TITLE
complete black after clear all tab am data

### ISSUE THREAD

**Issue by stevenleeS0ht** [duckduckgo/Android#1159]
**Describe the bug**
The app directly turn to black and not responsive after clear all tab and data.

**How to Reproduce**
1. Touch button clear all tab and data.
2. See error

If applicable, please include information about any settings that may be relevant to this issue (e.g. selected theme, auto clear option, etc).

**Expected behavior**

**Smartphone (please complete the following information):**
 - DDG App Version: 5.78.1 zw
 - Device: XIAOMI
 - OS: Android One 11
 - If applicable, add screenshots to help explain your problem.

**Comment by arunsathiya** [comment]
I see the same issue, has been happening for a few months but never had a chance to submit the bug report. My workaround at the moment is to close the app from "recent apps" list of Android and re-open it.

**Comment by stevenleeS0ht** [comment]
@CDRussell  and @subsymbolic , do you have any ideas on this issue?

**Comment by stevenleeS0ht** [comment]
@CDRussell , yes, when black the only thing I can do is quit app. The screen is completely dark.

**Comment by arunsathiya** [comment]
https://user-images.githubusercontent.com/18581859/113480953-a056cf80-94b4-11eb-8c29-ad3508056ecd.mp4

@CDRussell Here we go. We can see that I am making some searches on DuckDuckGo and then clear all data, but the app blacks out and I can scroll down to see the actual page in some sort of a layered background. I open the recent apps list to clear DuckDuckGo from it, and then re-launch the app.

**Comment by cmonfortep** [comment]
Thanks for all the info here.

We still don't understand why it fails. At that point where the screen turns black there should be an animation, so I guess the root cause should be related to that part. We will try to look into this issue and try to find a way to mitigate the issue for those cases where the animation fails to appear.

@stevenleeS0ht do you have animations disabled on your device by any chance? I guess the answer is no, but just double-checking this.

**Comment by arunsathiya** [comment]
@cmonfortep Good hint! I have `Window animation scale`, `Transition animation scale` and `Animator duration scale` turned off under `Developer options` on my Android settings. By just updating `Animator duration scale` to 1x, the fire animation appears okay on DuckDuckGo app. It would be great if this bug is resolved. Thanks!

**Comment by stevenleeS0ht** [comment]
@cmonfortep , Yes, developer mode enable, all animation turn off. Is it possible not to show the animation when the animation in setting are disabled?

**Comment by cmonfortep** [comment]
Yes, I guess we can try that strategy.

- check if animation settings are disabled
- if so, skip the animation (do the same as when user selected "none" animation in settings)

There's already some code inside `FireDialog.kt` to handle the scenario where the user didn't select any animation inside our settings, so it will just a matter of doing the same for that one.

It will be nice also to block gestures when clearing data, I've noticed in the video that users were able to scroll down the black screen. But I would leave that as optional.

Now that the approach is pretty clear, I will open this issue for possible contributions.
